app/domain/operations/hbx_enrollments/expire.rb

```
@@ -4,38 +4,193 @@ module Operations
+          return Failure(msg)
+        end
+
+        if params[:transmittable_identifiers][:transmission_gid].blank?
+          msg = "Missing transmission_gid in transmittable_identifiers of params: #{params}."
+          return Failure(msg)
+        end
+
+        if params[:transmittable_identifiers][:transaction_gid].blank?
+          msg = "Missing transaction_gid in transmittable_identifiers of params: #{params}."
+          return Failure(msg)
+        end
+
+        if params[:transmittable_identifiers][:subject_gid].blank?
+          msg = "Missing subject_gid in transmittable_identifiers of params: #{params}."
+          return Failure(msg)
+        end
+
+        Success(params)
+      end
+
+      def find_enrollment(enrollment_gid)
+        hbx_enrollment = GlobalID::Locator.locate(enrollment_gid)
+
+        if hbx_enrollment.blank?
+          msg = "No HbxEnrollment found with given global ID: #{enrollment_gid}"
+          Failure(msg)
+        elsif !hbx_enrollment.is_ivl_by_kind?
+          msg = "Failed to expire enrollment hbx id #{hbx_enrollment.hbx_id} - #{hbx_enrollment.kind} is not a valid IVL enrollment kind"
+          Failure(msg)
+        else
+          Success(hbx_enrollment)
+        end
       end
 
-      def find_enrollment(enrollment_hbx_id)
-        Operations::HbxEnrollments::Find.new.call({hbx_id: enrollment_hbx_id})
+      def find_and_update_request_transmission(transmission_gid, job)
+        transmission = find_transmission_by_global_id(transmission_gid)
+
+        if transmission.success?
+          status_result = update_status("Transmittable::Transmission found with given global ID: #{transmission_gid}",
+                                        :succeeded,
+                                        { transmission: transmission.value! })
+
+          return status_result if status_result.failure?
+          Success(transmission.value!)
+        else
+          add_errors(:invalid_transmission_on_enrollment_expire,
```

@saipraveen18 I think we should simply return a failure monad if we are unable to find the Transmission. This operation can be called with invalid params and we should not add errors to the job. This specific step should be split into 2 steps.
- Find the transmission by global_id
- Updating the Status of Transmission